velox/functions/prestosql/Comparisons.h

```diff
@@ -99,6 +116,10 @@ struct EqFunction {
   // Used for primitive inputs.
   template <typename TInput>
   void call(bool& out, const TInput& lhs, const TInput& rhs) {
+    if constexpr (std::is_floating_point_v<TInput>) {
+      out = util::floating_point::NaNAwareEquals<TInput>{}(lhs, rhs);
+      return;
+    }
```

Comment:
Do we have unit tests for these simple functions being changed? I imagine the added tests in ComparisionsTest.cpp are for the vector function?

Reply:
Also, does DistinctFromFunction need to be changed too?

Reply:
Yes, DistinctFromFunction also needs to change because `NaN isDistinctFrom NaN` should be false.

Reply:
> Do we have unit tests for these simple functions being changed? I imagine the added tests in ComparisionsTest.cpp are for the vector function?

Good point, is there a simpler way of switching between the 2 to be able to re-use this test for both?